config_test.go

```diff
@@ -52,3 +53,67 @@ func TestConfigToFlatMap(t *testing.T) {
 	require.Equal(t, redactedPassword, result["db.etcd.pass"])
 	require.Equal(t, redactedPassword, result["db.postgres.dsn"])
 }
+
+func TestSupplyEnvValue(t *testing.T) {
+	// Mock environment variables for testing
+	os.Setenv("EXISTING_VAR", "existing_value")
+	os.Setenv("EMPTY_VAR", "")
```

Comment:
Should use `t.Setenv()` instead, so they're cleaned up automatically at the end of the test.